packages/next/shared/lib/utils.ts

```diff
@@ -178,6 +179,7 @@ export type AppPropsType<
 
 export type DocumentContext = NextPageContext & {
   renderPage: RenderPage
+  jsxStyleRegistry: StyleRegistry
```

Comment:
Hmm, I don't think we want this exposed as part of the public API though... what if you made this:

```
defaultGetInitialProps: (context: DocumentContext) => DocumentInitialProps | Promise<DocumentInitialProps>
```

then changed `pages/_document.tsx` to just defer and return `ctx.defaultGetInitialProps(ctx)`, and then just passed in a version from `server/render.ts` that does what you have `pages/_document.tsx` doing now?